### PR TITLE
Implement tree assessor factory functions

### LINKED ISSUE #2126: Create factory functions to construct different types of assessors
After the work in #2123, #2124 and #2125 we should create a bunch of utility functions to create the different types of assessors we currently have in the code. This can simply be a file that exports all these functions as `export function constructAssessor() {}`. All these functions have the same arguments as the `TreeAssessor` itself. All these functions return a new instance of a `TreeAssessor`.

We require the following functions:

* `constructSEOAssessor()`, mimics the current `SEOAssessor` so copy the list of assessments that the `SEOAssessor` has. This assessor uses the `SEOScoreAggregator`.
* `constructReadabilityAssessor()`, mimics the current `ContentAssessor`, so copy the list of assessment that the `ContentAssessor` has. This assessor uses the `ContentScoreAggregator`.
* `constructTaxonomyAssessor()`, ☝️, uses `SEOScoreAggregator`.
* `constructRelatedKeywordAssessor()`, ☝️, uses `SEOScoreAggregator`.
* `constructRelatedKeywordTaxonomyAssessor()`, ☝️, uses `SEOScoreAggregator`.

Put these in a `cornerstone.js` folder to prevent making the names unwieldy:

* `constructReadabilityAssessor()`, ☝️, uses `ContentScoreAggregator`.
* `constructRelatedKeywordAssessor()`, ☝️, uses `SEOScoreAggregator`.
* `constructSEOAssessor()`, ☝️, uses `SEOScoreAggregator`.

If you know of any assessors I missed, please address them too.

Requires #2123 
Requires #2124 
Requires #2125

Parent #1903

### PR DESCRIPTION
**Pull request by hansjovis**
## Summary

This PR can be summarized in the following changelog entry:

* _[non-user-facing]_ Implements a collection of factory functions that create `TreeAssessor`s for different kinds of SEO and readability analysis. The kind produced depends on factors like type of content (normal page/post, cornerstone, taxonomy page) and type of keyphrase (main or related).

## Relevant technical choices:
I made some JSDoc submodules to keep relevant content together, whilst keeping them from all getting on the `tree/assess` module. I am not quite sure about the naming, so a second opinion would be nice!

* The factory functions live in their own JSDoc module called `tree/assess/assessors`.
     * Cornerstone functions live in `tree/assess/assessors/cornerstone`.
* I separated creating the assessment lists into separate factory functions, since they can grow quite big. They live in the submodule `tree/assess/assessors/assessments` and `tree/assess/assessors/cornerstone/assessments` respectively.

## Test instructions

This PR can be tested by following these steps:

* Run `yarn test`, check if the coverage is 100% for the `tree/assess` folder.
* Check if the tests cover all edge cases.
* Review the code a second time, check if you can find any missing functionality or code improvements.

Fixes #2126

### CODE REVIEW

File: src/tree/assess/assessmentListFactories.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ * @returns {module:tree/assess.Assessment[]} The list of SEO assessments.
+ *
+ * @memberOf module:tree/assess/assessors/assessments
+ */
+const constructTaxonomyAssessments = () => [
+	// Needs to be populated by fancy new assessments that work on the tree representation of the text.
+];
+
+/**
+ * Creates a new list of SEO assessments for taxonomy pages.
```
Comment (nataliashitova) [inline src/tree/assess/assessmentListFactories.js]:
```suggestion
 * Creates a new list of SEO assessments for related keyphrases.
```

File: src/tree/assess/assessmentListFactories.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ * @returns {module:tree/assess.Assessment[]} The list of SEO assessments.
+ *
+ * @memberOf module:tree/assess/assessors/assessments
+ */
+const constructRelatedKeyphraseAssessments = () => [
+	// Needs to be populated by fancy new assessments that work on the tree representation of the text.
+];
+
+/**
+ * Creates a new list of SEO assessments for taxonomy pages.
```
Comment (nataliashitova) [inline src/tree/assess/assessmentListFactories.js]:
```suggestion
 * Creates a new list of SEO assessments for related keyphrases on taxonomy pages.
```

File: src/tree/assess/cornerstone/assessmentListFactories.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ * @returns {module:tree/assess.Assessment[]} The list of readability assessments.
+ *
+ * @memberOf module:tree/assess/assessors/cornerstone/assessments
+ */
+const constructReadabilityAssessments = () => [
+	// Needs to be populated by fancy new assessments that work on the tree representation of the text.
+];
+
+/**
+ * Creates a new list of SEO assessments for taxonomy pages.
```
Comment (nataliashitova) [inline src/tree/assess/cornerstone/assessmentListFactories.js]:
```suggestion
 * Creates a new list of SEO assessments for related keyphrases.
```

**Review by nataliashitova — COMMENTED** [review]
Looks good to me 👍 
I am curious if with the new way assessment lists being created separately from the actual assessor constructors, we can optimize the latter. For instance, if we pass a config with ±cornerstone, ±related keyphrase and ±taxonomy to the assessor constructor, 8 functions (5 regular and 3 cornerstone) will become one.

**Comment by nataliashitova** [comment]
This PR has been transferred to the new monorepo https://github.com/Yoast/javascript/pull/6.
